src/qt3utils/datagenerators/princeton.py

```diff
@@ -0,0 +1,337 @@
+        self.light = LightfieldApplicationManager(True)
+
+    def finalize(self) -> None:
+        """
+        Closes the Lightfield application without saving the settings.
+        """
+        self.light.close()
+        
+    def stop_scan(self) -> None:
+        """ 
+        Stop/Pause the current acquisition. If you click "Start" you will be able to continue the scan.
+        """
+        self.light.experiment.Stop()
+    
+    def get_wavelengths(self) -> np.ndarray:
+        """
+        Returns the wavelength calibration for a single frame.
+
+        According to the person who made the Lightfield sofware there is not current way to use the exact Lightfield settings
+        for step and glue. Will have to interpolate for now.
+        """
+        wavelength_array = np.fromiter(self.light.experiment.SystemColumnCalibration, np.float64)
+        return wavelength_array
+    
+    @property
+    def center_wavelength(self) -> float:
+        """
+        Returns the center wavelength in nanometers.
+        """
+        return self.light.get(lf.AddIns.SpectrometerSettings.GratingCenterWavelength)
+
+    @center_wavelength.setter
+    def center_wavelength(self, nanometers: float) -> None:
+        """
+        Sets the spectrometer center wavelength to nanometers.
+        """
+        # The line below addresses bug where if step and glue is enabled it wont allow you to set the center wavelength.
+        self.light.set(lf.AddIns.ExperimentSettings.StepAndGlueEnabled, False)
+        self.light.set(lf.AddIns.SpectrometerSettings.GratingCenterWavelength, Double(nanometers))
+
+    @property
+    def experiment_name(self) -> Union[str, None]:
+        """
+        Returns the experiment name.
+        """
+        return self._experiment_name
+
+    @experiment_name.setter
+    def experiment_name(self, a_name: str) -> None:
+        """
+        User can set the experiment that they want to load.
+        """
+        if a_name != self._experiment_name:
+            # Added this to prevent errors with incorrect file name
+            if a_name in self.light.experiment.GetSavedExperiments(): 
+                self._experiment_name = a_name
+                self.light.load_experiment(self._experiment_name)
+            else:
+                logger.error(f"An experiment with that file name does not exist.")
+
+    @property
+    def grating(self) -> str:
+        """
+        Returns the current grating.
+        """
+        return self.light.get(lf.AddIns.SpectrometerSettings.GratingSelected)
+
+    @grating.setter
+    def grating(self, value: str) -> None:
+        """
+        Sets the current grating to be the one specified by parameter grating.
+        """
+        if value in self.grating_options:
+            self.light.set(lf.AddIns.SpectrometerSettings.GratingSelected, String(value))
+        else:
+            logger.error(f"Grating {value} is not an options. The options are: {self.grating_options}")
```

Comment:
maybe change name to selected_grating?

Reply:
Or make it match the lightfield values exactly except with snake case. `grating_selected'

ps: There's actually a nifty way we could essentially expose *all* of the lightfield settings with very little code. https://github.com/qt3uw/qcsapphire/blob/main/src/qcsapphire/pulser.py#L28 and read the doc here: https://github.com/qt3uw/qcsapphire/blob/main/README.md#property-like-calls